### PR TITLE
cli: improve the debug pebble db excise command

### LINKED ISSUE #122059: storage: add a pebble subcommand to excise a table
As a fallback solution for #118283 we want a way to use the CLI to excise an entire table, which would be used to undo a restore that attached external objects which are no longer accessible.

This doesn't affect normal production code and it's acceptable if it's in a 24.1.x release, so it's not a blocker.

Jira issue: CRDB-37682

### PR DESCRIPTION
**Pull request by RaduBerinde**
This commit makes the recently added `pebble db excise` command more
user-friendly: we allow excising an entire table by specifying the
table and tenant IDs. Example:

```
./cockroach debug pebble db excise cockroach-data --table-id=4 --tenant-id=2
Excising range:
  start: /Tenant/2/Table/4/0,0
  end:   /Tenant/2/Table/5/0,0
WARNING!!!
This command will remove all keys in this range!
Continue? [Y/N] y
Excise complete.
```

Fixes: #122059
Release note: None

### CODE REVIEW

File: pkg/cli/debug.go [hunk 1]
```diff
@@ -1631,10 +1634,37 @@ func initPebbleCmds(cmd *cobra.Command, pebbleTool *tool.T) {
...
 	}
 }
 
+// pebbleExciseSpanFn implements tool.DBExciseSpanFn. It returns a span
+// for a table if the table/tenant ID flags are used.
+func pebbleExciseSpanFn() (pebble.KeyRange, error) {
+	if debugPebbleOpts.exciseTenantID == 0 && debugPebbleOpts.exciseTableID == 0 {
+		return pebble.KeyRange{}, nil
+	}
+	if debugPebbleOpts.exciseTenantID == 0 || debugPebbleOpts.exciseTableID == 0 {
```
Comment (itsbilal) [inline pkg/cli/debug.go:1652]: There might also be value in blowing up the whole tenant, right? in which case we should be able to specify just the tenant ID and not the table ID?

**Review by itsbilal — APPROVED** [review]
LGTM

File: pkg/cli/debug.go (hunk 1 above)
Comment (RaduBerinde) [reply pkg/cli/debug.go:1652]: CC @msbutler WDYT?

Comment (msbutler) [reply pkg/cli/debug.go:1652]: yeah, this could be nice in a UA cluster, if we needed to wipe the main tenant.

Comment (RaduBerinde) [reply pkg/cli/debug.go]: Done. I added an `--entire-tenant` flag which must be used in conjunction with `--tenant-id` (to make accidents less likely). We disallow wiping the entire system tenant.

**Review by itsbilal — APPROVED** [review]
LGTM

**Comment by RaduBerinde** [comment]
TFTR!

bors r+

**Pull request merged**